dbt/adapters/redshift/connections.py

```diff
@@ -99,9 +116,27 @@ def get_connect_method(self):
             "port": self.credentials.port if self.credentials.port else 5439,
             "auto_create": self.credentials.autocreate,
             "db_groups": self.credentials.db_groups,
-            "region": self.credentials.host.split(".")[2],
+            "region": self.credentials.region,
             "timeout": self.credentials.connect_timeout,
         }
+        if kwargs["region"] is None:
+            logger.debug("No region provided, attempting to determine from host.")
+            try:
+                region_value = self.credentials.host.split(".")[2]
```

Comment:
I guess we've been running with this for a while add an earlier line – is this the pattern that red hift is going to stick to long-term?

Reply:
redshift does have a region field in the connector, it doesn't use it underneath the hood though.
[Connection invocation](https://github.com/aws/amazon-redshift-python-driver/blob/910be71314229e5b2febc60ddf8b8bcc992ea5f0/redshift_connector/__init__.py#L340)

Reply:
Oh that's so interesting and unexpected – I suppose we'll know if they change that interface, if an exception gets flagged here due to bad string processing

Reply:
>redshift does have a region field in the connector, it doesn't use it underneath the hood though.
[Connection invocation](https://github.com/aws/amazon-redshift-python-driver/blob/910be71314229e5b2febc60ddf8b8bcc992ea5f0/redshift_connector/__init__.py#L340)

If `redshift_connector` doesn't even use `region` under the hood, could we just always pass in `region=None`?

i.e., if the connector doesn't need it, then we don't need to collect it or parse it, validate it, etc.

Reply:
@dbeatty10 I don't think we can pass region as `None` always since it's used in IAM authn (as specified [here](https://github.com/aws/amazon-redshift-python-driver#example-using-iam-credentials) but not in password authn. If not specified I believe it's implicitly picked from the `~/.aws/config` file.

Reply:
Thanks for that info @sathiish-kumar 👍 

Part of the reason I asked is because it would be nice if we didn't need to add `region` option to the dbt profile options listed [here](https://docs.getdbt.com/reference/warehouse-setups/redshift-setup#authentication-methods) if we don't have to.

Two questions that are meant to be linked together in the same thought:
- Would it be possible to skip supplying the `region` when the user is using [password-based authentication](https://docs.getdbt.com/reference/warehouse-setups/redshift-setup#password-based-authentication)?
- And when using [IAM Authentication](https://docs.getdbt.com/reference/warehouse-setups/redshift-setup#iam-authentication), `redshift_connector` will implicitly get the `region` from the user's profile defined in `~/.aws/...`?